core/src/test/java/org/elasticsearch/index/analysis/AnalysisRegistryTests.java

```diff
@@ -144,51 +145,74 @@ public void testBackCompatOverrideDefaultIndexAndSearchAnalyzer() {
 
+    /**
+     * Tests that {@code camelCase} filter names and {@code snake_case} filter names don't collide.
+     */
     public void testConfigureCamelCaseTokenFilter() throws IOException {
         Settings settings = Settings.builder().put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString()).build();
         Settings indexSettings = Settings.builder()
                 .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-                .put("index.analysis.filter.wordDelimiter.type", "word_delimiter")
-                .put("index.analysis.filter.wordDelimiter.split_on_numerics", false)
-                .put("index.analysis.analyzer.custom_analyzer.tokenizer", "whitespace")
-                .putArray("index.analysis.analyzer.custom_analyzer.filter", "lowercase", "wordDelimiter")
-                .put("index.analysis.analyzer.custom_analyzer_1.tokenizer", "whitespace")
-                .putArray("index.analysis.analyzer.custom_analyzer_1.filter", "lowercase", "word_delimiter").build();
+                .put("index.analysis.filter.testFilter.type", "mock")
+                .put("index.analysis.filter.test_filter.type", "mock")
+                .put("index.analysis.analyzer.custom_analyzer_with_camel_case.tokenizer", "standard")
+                .putArray("index.analysis.analyzer.custom_analyzer_with_camel_case.filter", "lowercase", "testFilter")
+                .put("index.analysis.analyzer.custom_analyzer_with_snake_case.tokenizer", "standard")
+                .putArray("index.analysis.analyzer.custom_analyzer_with_snake_case.filter", "lowercase", "test_filter").build();
 
         IndexSettings idxSettings = IndexSettingsModule.newIndexSettings("index", indexSettings);
 
-        IndexAnalyzers indexAnalyzers = new AnalysisModule(new Environment(settings), emptyList()).getAnalysisRegistry()
+        /* The snake_case version of the name should not filter out any stopwords while the
+         * camelCase version will filter out English stopwords. */
+        AnalysisPlugin plugin = new AnalysisPlugin() {
+            class MockFactory extends AbstractTokenFilterFactory {
+                MockFactory(IndexSettings indexSettings, Environment env, String name, Settings settings) {
+                    super(indexSettings, name, settings);
+                }
+
+                @Override
+                public TokenStream create(TokenStream tokenStream) {
+                    if (name().equals("test_filter")) {
+                        return new MockTokenFilter(tokenStream, MockTokenFilter.EMPTY_STOPSET);
+                    }
+                    return new MockTokenFilter(tokenStream, MockTokenFilter.ENGLISH_STOPSET);
+                }
+            }
+            @Override
+            public Map<String, AnalysisProvider<TokenFilterFactory>> getTokenFilters() {
+                Map<String, AnalysisProvider<TokenFilterFactory>> filters = new HashMap<>();
+                filters.put("mock", MockFactory::new);
```

Comment:
I guess so! I think I wrote this like this at first because I thought I'd be adding more. I'm not sure, honestly. I'll change it.